### DISCUSSION TITLE
Can't call user summary after auth

### DISCUSSION THREAD

**Discussion by emreond** [sbertix/Swiftagram#233] [Q&A] [answered]
Hello,

I am trying to get user summary after authenticate with WebView. Webview is opening without any error and I am typing my pass and then It stucks at white screen and It doesn't get into userSummary calls completion blocks.

P.S: Authenticator succesfully returns secret.

```
                Authenticator.keychain
                .visual(filling: self.webView)
                .authenticate()
                .sink(receiveCompletion: { _ in }, receiveValue: { secret in
                     debugPrint("It succesfully returns secret")
                    Endpoint.user(secret.identifier)
                        .summary
                        .unlock(with: secret)
                        .session(.instagram)
                        .sink(receiveCompletion: { subscriber in
                            debugPrint(subscriber)
                          ....
                        }, receiveValue: { [weak self] user in
                            guard let user = user.user else { return }
                            ....
                        })
                        .store(in: &bin)
                })
                .store(in: &bin)
```

**Comment by sbertix** [comment, marked as answer]
Hey @emreond, 
are you perhaps on Xcode 13? `Authenticator.keychain` does not currently work with Xcode 13 (the dependency hasn't been updated to fix it yet, unfortunately). 

Otherwise do you mind telling me a bit more? Does the `Secret` `receiveValue` actually get called? Are you making sure you're retaining `bin`?

**Reply by emreond** [reply]
Nope I am on Xcode 12.5. Authenticator successfully returns secret. Therefore, I suppose, endpoint is get called but noting returns even with ` .session(.ephemeral, retries: 1, logging: .default)` . I assigned bin inside function just above `DispatchQueue` maybe It is because of that, I will try to make it global.

**Reply by sbertix** [reply]
It's probably because you're using Combine `Publisher`'s wrong, then. 
Try this, instead:

```swift
Authenticator.keychain
    .visual(filling: self.webView)
    .authenticate()
    .flatMap { secret in
        Endpoint.user(secret.identifier)
            .summary
            .unlock(with: secret)
            .session(.instagram)
    }
    .sink { _ in
    } receiveValue: {
        guard let user = $0.user else { return }
        print(user)
    }
    .store(in: &bin)
```

Keep in mind this flow, when you immediately rely on the logged in `Secret` to drive an `Endpoint`, is not ideal. 

If this does not work still, then it's because `bin` is not retained correctly and it's deallocated before it can actually publish an output. 
Please refer to **Combine** tutorials and documentation to learn more as this is not an issue related with **Swiftagram**.

**Reply by emreond** [reply]
This is working like a charm. I am not used to `Combine` , I need to check more. Thank u so much.
